### PR TITLE
feat: added wishlist management functionality

### PR DESCRIPTION
**Pull request by SanjayJatti**
- Add to wishlist functionality from product listing page
- Delete from wishlist functionality 
- Updating count in wishlist badge at header

Files to review:
- header.jsx 
- wishlistContext.jsx 
- wishlistCard.jsx 
- wishlist.jsx 
- wishlistFunc.js 
- productCard.jsx
- index.js

wishlist-management

### CODE REVIEW

**Review by SukanyaSen530 — APPROVED** [review]
Few suggestion -
1. The hover effect can be smooth - you can use transition property
2. Later you can add guest login feature 
3.  Add a basic loading state as the page loads but the images take time to load in product listing page.

File: src/Components/Wishlist/Wishlist.jsx [hunk 1]
```diff
@@ -1,7 +1,30 @@
-import React from 'react'
+import React from "react";
+import "./Wishlist.css";
+import { Header } from "../Header/Header.jsx";
+import { WishlistCard } from "./WishlistCard.jsx";
```
Comment (t007rushi) [inline src/Components/Wishlist/Wishlist.jsx:4]: can make single index file for imports

File: src/Components/Wishlist/WishlistCard.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            className="img-responsive"
+            src={product.image}
+            alt={product.title}
+          />
+          <div className="card-info">
+            <h4>{product.title}</h4>
+            <p>
+              {product.categoryName}{" "}
+              <span className="text-center">
+                {product.rating}
```
Comment (t007rushi) [inline src/Components/Wishlist/WishlistCard.jsx:29]: can destructure in fun. parameters itself

File: src/Context/WishlistContext.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+const WishlistProvider = ({ children }) => {
+  const [wishlist, setWishlist] = useState([]);
+  const { authState } = useAuth();
+  const { userInfo, token } = authState;
+
+  useEffect(() => {
+    const getWishlist = async () => {
+      if (userInfo.isUserLoggedIn) {
+        try {
+          const response = await axios.get("/api/user/wishlist", {
```
Comment (t007rushi) [inline src/Context/WishlistContext.jsx:16]: you can destructure {data} for redability

**Review by t007rushi — APPROVED** [review]
Make a hard code guest login 
liked the name

**Comment by SanjayJatti** [comment]
> Few suggestion -
> 
> 1. The hover effect can be smooth - you can use transition property
> 2. Later you can add guest login feature
> 3. Add a basic loading state as the page loads but the images take time to load in product listing page.

Added guest log in as per suggestion f6becaee7ce969a6a99b7f3ef380c8b6654fc4f3

**Comment by SanjayJatti** [comment]
> Make a hard code guest login liked the name

Added guest log in as per suggestion f6becaee7ce969a6a99b7f3ef380c8b6654fc4f3

**Pull request merged**